### PR TITLE
Encrypt/Decrypt using Rijndael Key in C#

### LINKED ISSUE #4182: [Languages] Encrypt/Decrypt using Rijndael Key in C#
#### Proposed title of article
Encrypt/Decrypt using Rijndael Key in C#
#### Proposed article introduction
I am going to Create a Rijndael object with the specified key and IV. Then decrypt to perform the stream transform and create the streams used for encryption. Using RijndaelManaged write all data to the stream and return the encrypted bytes from the memory stream. Using if condition statements check the arguments.
 how to decrypt/encrypt strings using the Rijndael algorithm will be discussed in details. The key is generated out of a password. I have to write an application which encrypts a personal number using the AES/Rijndael method.
#### Key takeaways 
1. Introduction to and working with Rijndael key
2.Examples to decrypt with Rijndael key
-string Text
-byte[ ] keyBytes
-byte[ ] VectorBytes
3.Example to encrypt with rijnadel key.
 -string Text
-byte[ ] key
-byte[ ]VectorBytes
4.Advantages of using Rijndael  key in c#.
5.Conclusion.
#### Article quality
 I intend to incorporate programme-centric concepts of encryption and decryption using Rijndael key that have been overlooked by other writers who have tried writing about these topics. My content will also be original and written in a format that a beginner shall understand what is discussed and the information shall remain useful and relevant to high end programmers.

#### References
 N/A

### PR DESCRIPTION
**Pull request by Bikorir**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #4182

### CODE REVIEW

**Comment by mohamedgh16** [comment]
@Bikorir Before we begin the review, please delete the duplicated copies of your pr inside the index.md file.

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by Bikorir** [comment]
Hi@mohamedgh16, I have removed the duplicated copies in the index.md.

File: content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+- [Asymmetric and symmetric encryption](#asymmetric-and-symmetric-encryption)
+- [Conclusion](#conclusion)
+
+###  Why encrypt and decrypt data?
+Below are reasons why we should encrypt and decrypt data:
+
+1. These are critical communication measures designed to ensure correct receiving and information processing.
+2. Data confidentiality is achieved through encryption
+3. A file owner can give or deny access to private data allowing for transmission of confidential data.
+4. Decryption allows the conversion of encrypted data to the original information as long as the decryptor has the original key
```
Comment (mohamedgh16) [inline content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md:24]: Missing "."

File: content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        }
+        static void Main(string[] args)
+        {
+            string doneencryption = Encode();
+            Console.Write(doneencryption);
+        }
+    }
+}
+```
+
```
Comment (mohamedgh16) [inline content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md:79]: Remove extra space lines. 
(implement this to the entire tutorial).

File: content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+```bash
+section engineering education
+```
+
+The encrypted string `bIwl1Y/220KPke+MzXxKErqBT5jkOEbDJnVEPxCzs7I=` is converted to its original meaning `section engineering education` using `publickey` and `secret key` keys.
+
+A `symmetric key` is used to encrypt data and can also be used to decrypt data using only one string.
+
+Symmetric keys known as `Symmetric Cryptography Algorithms` are utilized in data encryption and decryption. Symmetric Cryptography Algorithms use one key and store large data volumes, thus appropriate for use in DES and Rijndael platforms. `Asymmetric keys` use two keys for encryption and decryption(private and public keys). The public key encrypts data, whereas the private key decrypts data. `RSA` and `digital` signatures` employ asymmetric encryption.
```
Comment (mohamedgh16) [inline content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md:150]: Try to enhance this section, also use the code snippets correctly (` employ asymmetric encryption)

File: content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+```C#
+textBoxString.Text = Encrypt.DecryptString(textBoxEncrypted.Text, textBoxPassword.Text);
+```
+
+### Using encryption and decryption program
+The Rijndael algorithm is used to build a durable (i.e., non-random) symmetric key and use it to encrypt and decode a text string. Generation of a key requires a combination of characteristics of encryption and decryption.
+
+Symmetric keys are guaranteed to be the same as long as the encryption and decryption methods employ the same parameters to generate them. Static functions make showing encryption and decryption logic in a class easier using duplicate codes found in static function and decryption logic. 
+
+Rijndael class creates new objects, a key, and an initialization vector. When overridden in a derived class, the GenerateKey() function generates a random key () used in the algorithm and a random initialization vector. Bytes are decrypted and stored as a string and then encrypted to an array of bytes. The string should show both the encrypted and unencrypted data. When condition statements are used to check an argument, a `Rijndael object` is created using the key and Initialization Vector.
```
Comment (mohamedgh16) [inline content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md:173]: Fix any spelling errors here.

File: content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+1. Small ciphertext is used in the symmetric key, and the asymmetric process requires a larger ciphertext than the file used. 
+2. The size of data used in symmetric encryption carries extensive data size, whereas small data are employed in asymmetric encryption.
+3. Symmetric key lengths are 128 or 256-bit, whereas RSA 2048 and higher key bits are applied in asymmetric encryption.
+4. A single key makes symmetric encryption prone to eavesdroppers compared to asymmetric keys with double keys encryption and decryption.
+5. Symmetic encryption is a relatively new technique compared to asymmetric encryption, which is an older encryption technique.
+6. Symmetric encryption is a fast technique of encryption when compared to the slow speed experienced when using asymmetric encryption.
+7. Symmetric encryption uses AES, DES,3DES, and RC4 algorithms, whereas asymmetric encryption uses RSA and ECC algorithms.
+
+### Conclusion
+Converting data into unreadable form through encryption and reverting it to its original form through decryption is critical in keeping confidential data transmission.
```
Comment (mohamedgh16) [inline content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md:190]: Fix any spelling errors here.

File: content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+In this article, the basics of encryption using advanced encryption standards (AES) are discussed in depth. Rijndael encryption is the gold standard for low-security leaks, and understanding how to use it is pivotal in programming. 
+An encryption algorithm appropriate for the task should be used during any program handling.
```
Comment (mohamedgh16) [inline content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md:193]: Add a further reading section after the conclusion.

**Review by ghost — COMMENTED** [review]
Great tutorial @Bikorir , please implement the changes so we can move forward.

**Comment by Bikorir** [comment]
Hello, @mohamedgh16  Am sorry for taking a while to make the edits. I was writing the final exams which made my time tightly scheduled. Please go through the changes I have made when you are ready. Thank you.

**Comment by mohamedgh16** [comment]
@Bikorir Remove the usual includes in c# to make the code shorter.

**Comment by hectorkambow** [comment]
Jumping on this one

**Comment by hectorkambow** [comment]
👏 

Nicely done!


A skill section would greatly benefit your author profile - to better inform potential future employers. 🚀

**Pull request merged**